src/app/routes/connector-ui/policy-definition-page/new-policy-dialog/new-policy-dialog-form.ts

```diff
@@ -47,16 +49,28 @@ export class NewPolicyDialogForm {
         connectorId: ['', Validators.required],
       });
 
-    // Switch validation depending on selected policy type
-    switchDisabledControlsByField({
-      formGroup,
-      switchCtrl: formGroup.controls.policyType,
-      enabledControlsByValue: {
-        'Connector-Restricted-Usage': ['connectorId'],
-        'Time-Period-Restricted': ['range'],
+    switchDisabledControls<NewPolicyDialogFormValue>(
+      newPolicyFormGroup,
+      (value) => {
+        const timePeriodRestricted =
+          value.policyType === 'Time-Period-Restricted';
+        const startDateOnly = value.rangeIsOpenEnded!;
+        const dateRange = !value.rangeIsOpenEnded!;
```

Comment:
i can understand if you want the expressions at the bottom to be shorter. however, I don't think we need an entire variable for the negation. My suggestion:

```ts
const openEnded = value.rangeIsOpenEnded!;
// [...]

return {
  //  [...]
  rangeIsOpenEnded: timePeriodRestricted,
  rangeStart: timePeriodRestricted && openEnded,
  range: timePeriodRestricted && !openEnded,
  //  [...]
}
```